data/json/construction/floors_indoors.json

```
@@ -452,11 +452,14 @@
     "id": "constr_dirtfloor",
     "group": "build_dirt_floor",
     "category": "CONSTRUCT",
-    "required_skills": [ [ "fabrication", 3 ] ],
+    "required_skills": [ [ "survival", 1 ] ],
     "time": "15 m",
+    "qualities": [ { "id": "DIG", "level": 1 } ],
+    "pre_flags": [ "DIGGABLE", "PLOWABLE" ],
     "pre_note": "Must be supported on at least two sides.",
     "pre_special": "check_support",
```

```suggestion
    "pre_special": [ "check_support", "check_unblocked" ],
```
Suggestions:
Add `check_unblocked` so you can only build on empty tile.
Remove pre_flags `PLOWABLE`, otherwise you can't build dirt floor on terrains like [upturned dirt](https://cdda-guide.nornagon.net/terrain/t_dirtmound).